### PR TITLE
fix: update hardhat and hardhat-zksync-solc in e2e-tests package

### PR DESCRIPTION
**Pull request by Romsters**

### CODE REVIEW

File: e2e-tests/package.json [hunk 1]
```diff
@@ -8,7 +8,7 @@
   "devDependencies": {
     "@matterlabs/eslint-config-vue": "^1.1.4",
     "@matterlabs/hardhat-zksync-deploy": "^0.6.5",
-    "@matterlabs/hardhat-zksync-solc": "^0.4.2",
+    "@matterlabs/hardhat-zksync-solc": "^1.2.5",
```
Comment (dutterbutter) [inline e2e-tests/package.json:11]: We should just update fully here and use  `@matterlabs/hardhat-zksync` ? Thoughts?

File: e2e-tests/hardhat.config.ts [hunk 2]
```diff
@@ -6,7 +6,7 @@ import "@nomiclabs/hardhat-ethers";
 
 const config: HardhatUserConfig = {
   zksolc: {
-    version: "1.5.7",
+    version: "1.5.8",
```
Comment (dutterbutter) [inline e2e-tests/hardhat.config.ts:9]: Did not realize this was released yet 👀  Does this come with lib linking?

Comment (Romsters) [reply e2e-tests/hardhat.config.ts:9]: Not sure, but the reason for this PR is that 1.5.8 only works with latest @matterlabs/hardhat-zksync-solc

File: e2e-tests/package.json (hunk 1 above)
Comment (Romsters) [reply e2e-tests/package.json:11]: Yeah, let's do it

File: e2e-tests/package.json [hunk 3]
```diff
@@ -23,7 +22,7 @@
     "eslint-config-prettier": "^9.0.0",
     "eslint-plugin-mocha": "^10.2.0",
     "ethers": "^5.7.2",
```
Comment (dutterbutter) [inline e2e-tests/package.json:24]: If we are using `1.3.0` we should update to ethers v6. v5 uses < `1.0.0`.

Comment (Romsters) [reply e2e-tests/package.json]: and update all the tests to v6

**Comment by Romsters** [comment]
Closes #491

**Review by dutterbutter — APPROVED** [review]

**Pull request merged**